src/Adyen/HttpClient/CurlClient.php

```diff
@@ -202,7 +202,7 @@ public function requestPost(\Adyen\Service $service, $requestUrl, $params)
 
 		curl_close($ch);
 
-		if ($httpStatus != 200 && $result) {
+		if ($httpStatus >= 400 && $result) {
```

Comment:
```suggestion
		$resultOKHttpStatusCodes = array(200, 201, 202, 203, 204);
		
		if (!in_array($httpStatus, $resultOKHttpStatusCodes) && $result) {
```

I would keep it "simple" and consider OK only the known response codes that can be returned from the API.